packages/routers/src/StackRouter.tsx

```diff
@@ -396,6 +396,9 @@ export default function StackRouter(options: StackRouterOptions) {
             }
 
             if (index === -1 && action.payload.name !== undefined) {
+              if (!state.routeNames.includes(action.payload.name)) {
```

Comment:
I think it would be good to move this to the top of this case, and change to something like:

```js
if (action.payload.name !== undefined && !state.routeNames.includes(action.payload.name)) {
  return null;
}
```

Reply:
I am not sure if it won't any intended workflow other than that, what I did is I read the old code and added this if statement